src/NuGet.Core/NuGet.Protocol/Plugins/PluginDiscoveryUtility.cs

```diff
@@ -0,0 +1,59 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NuGet.Common;
+
+namespace NuGet.Protocol.Plugins
+{
+    public static class PluginDiscoveryUtility
+    {
+        public static string InternalPluginDiscoveryRoot { get; set; } = null;
+
+        public static string GetInternalPlugins()
+        {
+            var rootDirectory = InternalPluginDiscoveryRoot ?? System.Reflection.Assembly.GetEntryAssembly()?.Location;;
```

Comment:
while it's okay to be safe, will GetEntryAsssembly ever return null though? Being over safe sometimes masks errors

Reply:
Per https://msdn.microsoft.com/en-us/library/system.reflection.assembly.getentryassembly(v=vs.110).aspx

```
Can return null when called from unmanaged code.
```

While technically not going to happen in our case, if someone's using our libraries it could break them. 

Additionally, this will change once we have inserted the plugins in VS/dotnet CLI. I will call that out explicitly in the PR.